### PR TITLE
use 2018 edition syntax

### PR DESCRIPTION
**Pull request by zimond**
Hey, this PR changes every crate in swc to actually adopt to Rust's 2018 syntax.

### CODE REVIEW

File: ecmascript/parser/src/lexer/state.rs [hunk 1]
```diff
@@ -292,7 +292,7 @@ impl State {
     }
 
     fn update(&mut self, start: BytePos, next: &Token) {
-        trace!(
+        log::trace!(
```
Comment (kdy1) [inline ecmascript/parser/src/lexer/state.rs]: Please change this to trace! with use log::trace.

File: ecmascript/parser/src/lexer/state.rs [hunk 2]
```diff
@@ -544,14 +544,14 @@ impl TokenContexts {
     }
     pub fn pop(&mut self) -> Option<TokenContext> {
         let opt = self.0.pop();
-        trace!("context.pop({:?})", opt);
+        log::trace!("context.pop({:?})", opt);
```
Comment (kdy1) [inline ecmascript/parser/src/lexer/state.rs]: Please change this to trace! with use log::trace.

File: ecmascript/parser/src/lexer/state.rs [hunk 3]
```diff
@@ -544,14 +544,14 @@ impl TokenContexts {
...
         opt
     }
     pub fn current(&self) -> Option<TokenContext> {
         self.0.last().cloned()
     }
     fn push(&mut self, t: TokenContext) {
-        trace!("context.push({:?})", t);
+        log::trace!("context.push({:?})", t);
```
Comment (kdy1) [inline ecmascript/parser/src/lexer/state.rs]: Please change this to trace! with use log::trace.

File: ecmascript/codegen/macros/src/lib.rs [hunk 4]
```diff
@@ -57,26 +51,21 @@ fn (&mut self, node: Node) -> Result;
         };
 
         Quote::new_call_site()
-            .quote_with(smart_quote!(
+            .quote_with(pmutil::smart_quote!(
```
Comment (kdy1) [inline ecmascript/codegen/macros/src/lib.rs]: Can you change this to `smart_quote!` with `use pmutil::smart_quote;`?

**Comment by kdy1** [comment]
I prefer `util/mod.rs` over `util.rs` if util has submodules. Can you change them?

File: ecmascript/parser/src/lexer/number.rs [hunk 5]
```diff
@@ -244,7 +244,7 @@ impl<'a, I: Input> Lexer<'a, I> {
             "radix for read_int should be one of 2, 8, 10, 16, but got {}",
             radix
         );
-        trace!("read_digits(radix = {}), cur = {:?}", radix, self.cur());
+        log::trace!("read_digits(radix = {}), cur = {:?}", radix, self.cur());
```
Comment (kdy1) [inline ecmascript/parser/src/lexer/number.rs]: Please change this to trace! with use log::trace

File: ecmascript/parser/src/parser/macros.rs [hunk 6]
```diff
@@ -114,7 +114,7 @@ macro_rules! assert_and_bump {
 ///     if token has data like string.
 macro_rules! eat {
     ($p:expr, ';') => {{
-        trace!("eat(';'): cur={:?}", cur!($p, true));
+        log::trace!("eat(';'): cur={:?}", cur!($p, true));
```
Comment (kdy1) [inline ecmascript/parser/src/parser/macros.rs]: Please change this to trace! with use log::trace

File: ecmascript/codegen/macros/src/lib.rs (hunk 4 above)
Comment (zimond) [reply ecmascript/codegen/macros/src/lib.rs]: > I prefer `util/mod.rs` over `util.rs` if util has submodules. Can you change them?

This is a main feature of 2018 edition. It's designed to resolve `mod.rs` jumping out in every search result and it's hard to find out which module the `mod.rs` is actually pointing to. Actually it's the main reason of this PR.

If you prefer mod.rs, I think it's better to remove 2018 edition declaration and/or put this into `rustfmt.toml` as a hint for contributors

Comment (zimond) [reply ecmascript/codegen/macros/src/lib.rs]: oops, I replied in the wrong comment. But the content declares itself, I think.

Comment (kdy1) [reply ecmascript/codegen/macros/src/lib.rs]: > This is a main feature of 2018 edition. It's designed to resolve mod.rs jumping out in every search result and it's hard to find out which module the mod.rs is actually pointing to. Actually it's the main reason of this PR

Oh.. I didn't know that. I agree that mod.rs showing everywhere is bit annoying.
I'll just change my editor's config to align files based on named, not type.

Comment (zimond) [reply ecmascript/codegen/macros/src/lib.rs]: Removing `mod.rs` leaves many submodules containing only a test file. That's kind of weird indeed. In the future we could split the tests into integration tests (move them into the related module files) and unit tests(move to `/tests` dir). That will make the project cleaner.

File: macros/common/src/binder.rs [hunk 7]
```diff
@@ -119,7 +119,7 @@ impl<'a> VariantBinder<'a> {
     pub fn qual_path(&self) -> Path {
         match self.enum_name {
             Some(enum_name) => Quote::new(def_site::<Span>())
-                .quote_with(smart_quote!(
+                .quote_with(pmutil::smart_quote!(
```
Comment (kdy1) [inline macros/common/src/binder.rs]: Can you change this to smart_quote with use pmutil::smart_quote?

File: macros/common/src/derive/generics.rs [hunk 8]
```diff
@@ -97,7 +97,7 @@ impl<'a> Derive<'a> {
 
     /// Add `Self: #trait_`.
     pub fn bound_self(&mut self, trait_: Path) {
-        let self_ty: Type = parse(quote!(Self).into()).unwrap();
+        let self_ty: Type = parse(quote::quote!(Self).into()).unwrap();
```
Comment (kdy1) [inline macros/common/src/derive/generics.rs]: Can you change this to quote! with use quote::quote?

File: macros/common/src/syn_ext.rs [hunk 9]
```diff
@@ -60,11 +60,11 @@ impl ItemImplExt for ItemImpl {
         let mut item: ItemImpl = {
             let (impl_generics, ty_generics, where_clause) = generics.split_for_impl();
             let item = if let Some((ref polarity, ref path, ref for_token)) = self.trait_ {
-                quote! {
+                quote::quote! {
```
Comment (kdy1) [inline macros/common/src/syn_ext.rs]: Can you change this to quote with use quote::quote?

File: macros/common/src/syn_ext.rs [hunk 10]
```diff
@@ -60,11 +60,11 @@ impl ItemImplExt for ItemImpl {
...
                     impl #impl_generics #polarity #path #for_token #ty #ty_generics #where_clause {}
                 }
             } else {
-                quote! {
+                quote::quote! {
```
Comment (kdy1) [inline macros/common/src/syn_ext.rs]: Can you change this to quote with use quote::quote?

**Review by kdy1 — CHANGES_REQUESTED** [review]

**Review by kdy1 — APPROVED** [review]
Done. Thank you very much!

**Pull request merged**